include/ibamr/CIBStochasticMethod.h

```diff
@@ -0,0 +1,176 @@
+
+namespace mu
+{
+class Parser;
+} // namespace mu
+
+namespace IBTK
+{
+class HierarchyMathsOps;
+} // namespace IBTK
+namespace IBAMR
+{
+class CIBStandardInitializer;
+class CIBStaggeredStokesOperator;
+} // namespace IBAMR
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBAMR
+{
+/*!
+ * \brief Class CIBFEMethod is a concrete CIBStrategy and IBMethod
+ * class which implements the motion of rigid bodies using the constraint
+ * formulation. The immersed structure is discretized using standard IB
+ * markers.
+ */
+
+class CIBStochasticMethod : public IBAMR::CIBMethod
+{
+public:
+    /*!
+     * \brief Constructor of the class.
+     */
+    CIBStochasticMethod(const std::string& object_name,
+                        SAMRAI::tbox::Pointer<SAMRAI::tbox::Database> input_db,
+                        const int no_structures = 1,
+                        bool register_for_restart = true);
+
+    /*!
+     * \brief Destructor of the class.
+     */
+    ~CIBStochasticMethod();
+
+    /*!
+     * \brief setter for KT.
+     */
+    void setkT(double kT);
+
+    /*!
+     * \brief setter for Length scale.
+     */
+    void setLScale(double scale);
+
+    /*!
+     * \brief setter for RFD delta.
+     */
+    void setRFdelta(double scale);
+
+    /*!
+     * \brief getter for KT.
+     */
+    double getkT() const;
+
+    /*!
+     * \brief getter for Length scale.
+     */
+    double getLScale() const;
+
+    /*!
+     * \brief getter for RFD delta.
+     */
+    double getRFdelta() const;
+
+    void preprocessIntegrateData(double current_time, double new_time, int num_cycles);
+
+    /*!
+     * \brief Advance the positions of the Lagrangian structures
+     */
+    void midpointStep(double current_time, double new_time);
+
+    void checkLagUpdate();
+
+    /*!
+     * \brief Advance the positions of the Lagrangian structure by a small increment
+     */
+    void moveLagrangianData(double delta);
+
+    /*!
+     * \brief helper function for RFD
+     */
+    void computeRFDforcesAndDisplacements();
+
+    /*!
+     * \brief sets rigid body velocity of structures at the half-time positions
+     */
+    void setHalfTimeVelocity(Vec U);
+
+    /*!
+     * \brief resetsets rigid body velocity of structures at the half-time positions to those at current time
+     */
+    void resetRFDVelocity();
+
+    /*!
+     * \brief configuration getter
+     */
+    void getConfiguration(const unsigned int part, Eigen::Vector3d& C, Eigen::Quaterniond& Q);
+
+    /*!
+     * \brief initial configuration getter
+     */
+    void getInitialCOM(const unsigned int part, Eigen::Vector3d& C);
+
+    bool getReject();
+
+    int getNumReject();
+
+    //////////////////////////////////////////////////////////////////////////////
+
+protected:
+    //////////////////////////////////////////////////////////////////////////////
+
+private:
+    // Boltzman Constant
+    double d_kT, d_L_scale, d_rf_delta;
```

Comment:
The current convention is to use C++11-style initialization in the header. I will fix this.